File: two-pointers/remove-element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        int count = 0
        for (int i = 0; i < len(nums); i += 1){
            if (nums[i] != val){
                nums[count] = nums[i];
                count += 1;
            }
        }
        return count;

File: two-pointers/remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {

        // init
        int count = 0;

        // move valid elements to the front
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) {
                nums[count] = nums[i];
                count++;
            }
        }

        // return the number of valid elements
        return count;
    }
}